### PR TITLE
Test cases for code in docs

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

It adds tests for code in documentation

* **What is the current behavior?** (You can also link to an open issue here)

There are no tests for code in documentation

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

File: starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from pathlib import Path
+import pytest
+from starknet_py.tests.e2e.utils import DevnetClientFactory
+
+directory = os.path.dirname(__file__)
+map_source_code = Path(directory, "map.cairo").read_text("utf-8")
+
+
+@pytest.mark.asyncio
+async def test_using_account_client(run_devnet):
```
Comment (war-in) [inline starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py:11]: For an unknown reason, this test takes much more time than others. any ideas?

File: starknet_py/tests/e2e/docs/guide/test_deploying_new_contracts.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import pytest
+from starknet_py.tests.e2e.utils import DevnetClientFactory
+
+directory = os.path.dirname(__file__)
+
+
+@pytest.mark.asyncio
+async def test_deploying_new_contracts(run_devnet):
+    # pylint: disable=import-outside-toplevel, disable=duplicate-code
+    # add to docs: start
```
Comment (war-in) [inline starknet_py/tests/e2e/docs/guide/test_deploying_new_contracts.py:11]: I've marked all code that should be in the docs:

# add to docs: start
code
code
# add to docs: end

code for test

# add to docs: start
...

File: starknet_py/tests/e2e/docs/guide/test_using_existing_contracts.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "account", "type": "felt"}],
+        "name": "balanceOf",
+        "outputs": [{"name": "balance", "type": "felt"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+]
```
Comment (war-in) [inline starknet_py/tests/e2e/docs/guide/test_using_existing_contracts.py:25]: Is it a proper way of showing abi? "poe format" did that and I'm not sure

**Review by Arcticae — APPROVED** [review]

File: starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py (hunk 1 above)
Comment (cptartur) [reply starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py:11]: You're deploying an Account to testnet in line 19 which is much slower that devnet. Perhaps there's a way to somehow include this line in docs but skip actual deployment?

File: starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    k, v = 13, 4324
+    # Adds a transaction to mutate the state of k-v store. The call goes through account proxy, because we've used
+    # AccountClient to create the contract object
+    await (
+        await map_contract.functions["put"].invoke(k, v, max_fee=0)
+    ).wait_for_acceptance()
+
+    (resp,) = await map_contract.functions["get"].call(
+        k
+    )  # Retrieves the value, which is equal to 4324 in this case
```
Comment (cptartur) [inline starknet_py/tests/e2e/docs/quickstart/test_using_account_client.py:44]: Maybe move this comment above the await, so it's not hanging around single bracket

**Review by cptartur — COMMENTED** [review]
Good work! Finally we'll have proper tests for our docs.

Few comment's though 😅

**Pull request merged**